### PR TITLE
[dashboard] add button.gp-link CSS class

### PR DESCRIPTION
**Pull request by trumbitta**
## Description

Adds a `.gp-link` Tailwind class to make buttons mimic links and avoid using `<a href="javascript:void(0)" ...>`

To be used for anchor tags that should've been buttons because they don't really bring the user to any new location like an anchor tag usually does.

Just like #9269, this is about #3841 and its purpose is to simplify to the max #8995.

## Related Issue(s)

Re #3841

## How to test

- launch the dashboard app
- the two updated links which are now buttons should look and work just like before

## Release Notes

```release-note
- [dashboard]: Added a `.gp-link` Tailwind class to make buttons mimic anchor tags
```

### CODE REVIEW

**Comment by laushinka** [comment]
Thanks, @trumbitta! Let me loop in @gtsiolis here for this one :)

**Comment by gtsiolis** [comment]
Looking at this now! 👀 

/werft run

**Comment by gtsiolis** [comment]
@trumbitta could you rebase this to `main` branch to fix the failing build?

File: components/dashboard/src/index.css
Comment (trumbitta) [inline components/dashboard/src/index.css:65]: wait a minute... where does this extra white-space between `hover:` and `bg-transparent` come from 🤔 

fixing it now (also in the next selector)

Comment (trumbitta) [reply components/dashboard/src/index.css]: @gtsiolis should be fixed now. Sorry for the hiccup, I don't really know how it happened 😮

**Comment by trumbitta** [comment]
Is werft having a bad day?

File: components/dashboard/src/projects/NewProject.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/NewProject.tsx:497]: **praise**: Nice!

Comment (gtsiolis) [inline components/dashboard/src/projects/NewProject.tsx:497]: **suggestion(non-blocking)**: We have a few more actions in **/plans** masked as links that could benefit from this. However, I wouldn't block this from merging it as we're going to update and replace plans entirly soon.

https://github.com/gitpod-io/gitpod/blob/08954965c673d38e5aa042de9146bf5406929dd8/components/dashboard/src/settings/Plans.tsx#L390-L396

https://github.com/gitpod-io/gitpod/blob/08954965c673d38e5aa042de9146bf5406929dd8/components/dashboard/src/settings/Plans.tsx#L452-L458

...

File: components/dashboard/src/projects/NewProject.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/NewProject.tsx:372]: **issue(non-blocking)**: This will slightly affect how the link style is rendered but that's minor for now.

| BEFORE | AFTER |
|-|-|
| link-before | link-after |

**Review by gtsiolis — APPROVED** [review]
Thanks for opening this PR, your patience, and caring about accessibility with replacing these links with buttons, @trumbitta! 💯 

To add some background context here, the introduction of the `gp-link` class in https://github.com/gitpod-io/gitpod/pull/5112 initially aimed to abstract the style used for help links. 🔗 

However, the long-term goal is to have a button component _variant_ styled as a link for _actions_ like the ones changed in this PR, instead of maintaining CSS styles in the `index.css`. 💡 

Ideally, we'd like to introduce a button component for what we need here but I don't want to block or close this pull request since we're going to update and replace the pages affected here soon. 🚀 

UX LGTM!

@laushinka could you take another look and merge this since this requires also an approval from the WebApp team?

File: components/dashboard/src/projects/NewProject.tsx
Comment (trumbitta) [reply components/dashboard/src/projects/NewProject.tsx:372]: Good catch 🏅 !

I guess it stems from [`text-sm` in the Tailwind styles applied to the `button` element](https://github.com/gitpod-io/gitpod/blob/7b0acb66f47d47ee5533a8a6c3cba545a677272d/components/dashboard/src/index.css#L50) while `a` seems to just inherit as usual.

The upcoming button component should take it into account. If you need a volunteer for working on the component, you know where to find me 😊

**Comment by trumbitta** [comment]
> @trumbitta I need to stop apologizing to you and just review faster! This fell through the cracks of my reviewing cycle 🙇🏻‍♀️
> 
> LGTM

🙏  @laushinka I'm yet to find anything you should apologize for. If anything, I owe you because sometimes I just find bits of your code in the dashboard codebase that are so good I want to scream.

**Pull request merged**
